force-app/main/default/classes/JobApplicationStatusHelper.cls

```diff
@@ -0,0 +1,106 @@
+            createTask('Submit your application on the company website if possible', 'High', jobApp);
+        }
+
+        //Create the below Tasks when the Job Application Status is 'Applied'
+        when 'Applied' {
+            createTask('Reach out to the hiring manager or recruiter', 'High', jobApp);
+            createTask('Follow up on your application via email weekly', 'High', jobApp);
+            createTask('Continue identifying and saving similar job opportunities', 'High', jobApp);
+            createTask('Set up weekly networking calls to explore similar companies/roles', 'High', jobApp);
+        }
+
+        //Create the below Tasks when the Job Application Status is 'Applied'
+        when 'Interviewing' {
+            createTask('Prepare your blurb or “tell me about yourself” response', 'High', jobApp);
+            createTask('Practice answering behavioral interview questions', 'High', jobApp);
+            createTask('Research the company and your interviewers', 'High', jobApp);
+            createTask('Set up your virtual interview space and test your tech', 'High', jobApp);
+            createTask('Send thank you emails within 24 hours', 'High', jobApp);
+        }
+
+        //Create the below Tasks when the Job Application Status is 'Negotiating'
+        when 'Negotiating' {
+            createTask('Research your market value and know your numbers', 'High', jobApp);
+            createTask('Prepare your negotiation scripts', 'High', jobApp);
+            createTask('Evaluate your offer and decline or accept', 'High', jobApp);
+        }
+
+        ////Create the below Tasks when the Job Application Status is 'Accepted'
+
+        when 'Accepted' {
+            createTask('Plan your resignation if applicable', 'High', jobApp);
+            createTask('Take some time to relax and recharge', 'High', jobApp);
+            createTask('Prepare for your first day of onboarding', 'High', jobApp);
+        }
+
+        //Create the below Tasks when the Job Application Status is 'Closed'
+        when 'Closed' { 
+            createTask('Send a follow-up email thanking the interviewer and asking for feedback', 'High', jobApp);
+            createTask('Review your notes and reflect on areas of improvement', 'High', jobApp);
+        }
+    }
+}
+    //Method to create Tasks
+    private static void createTask(String description, String priority, Job_Application__c application){
+
+        Task task = new Task();
+        task.Description = description;
+        task.Subject = 'Action Required';
+        task.ActivityDate = System.today() + 1;
+        task.Priority = priority;
+        task.WhatId = application.Id;
+        task.OwnerId = application.OwnerId;
+        task.Status = 'In Progress';
+        //Add this task to the list of Tasks to be Created
+        tasksToBeCreated.add(task);
+    }
+
+    private static List<Task> tasksToBeCreated = new List<Task>();
```

Comment:
Move variable to top of class for visibility

Reply:
Moved to the top - Line 11